packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -484,8 +488,45 @@ export class FeatureDevController {
                 })
                 await session.updateChatAnswer(tabID, i18n('AWS.amazonq.featureDev.pillText.acceptAllChanges'))
                 await session.sendLinesOfCodeGeneratedTelemetry()
+                await session.sendMetricDataTelemetry(
+                    MetricDataOperationName.END_CODE_GENERATION,
+                    MetricDataResult.SUCCESS
+                )
             }
             this.messenger.sendUpdatePlaceholder(tabID, i18n('AWS.amazonq.featureDev.pillText.selectOption'))
+        } catch (err: any) {
+            getLogger().error(`${featureName}: Error during code generation: ${err}`)
+
+            switch (err.constructor.name) {
+                case FeatureDevServiceError.name:
+                    if (err.code === 'EmptyPatchException') {
+                        await session.sendMetricDataTelemetry(
+                            MetricDataOperationName.END_CODE_GENERATION,
+                            MetricDataResult.LLMFAILURE
+                        )
+                    } else {
+                        await session.sendMetricDataTelemetry(
+                            MetricDataOperationName.END_CODE_GENERATION,
+                            MetricDataResult.ERROR
```

Comment:
This is generic case right? Should this be a fault? Should we inspect for more codes?